src/util/astQuery.ts

```diff
@@ -0,0 +1,75 @@
+import * as ts from 'typescript';
+import {current} from './syntaxKind';
+import {Maybe, ifTrue} from './function';
+const kinds = current();
+
+export function isCallExpression(expr: ts.LeftHandSideExpression): expr is ts.CallExpression {
+    return expr && expr.kind === kinds.CallExpression;
+}
+
+export function callExpression(dec: ts.Decorator): Maybe<ts.CallExpression> {
+    return Maybe.lift(dec.expression)
+        .fmap(expr => isCallExpression(expr) ? expr as ts.CallExpression : undefined);
+}
+
+export function isPropertyAssignment(expr: ts.ObjectLiteralElement): expr is ts.PropertyAssignment {
+    return expr && expr.kind === kinds.PropertyAssignment;
+}
+
+export function isSimpleTemplateString(expr: ts.Expression): expr is (ts.StringLiteral | ts.NoSubstitutionTemplateLiteral) {
+    return expr && expr.kind === kinds.StringLiteral || expr.kind === kinds.NoSubstitutionTemplateLiteral;
```

Comment:
I believe https://github.com/mgechev/codelyzer/blob/master/src/util/utils.ts#L29 should use `kinds.NoSubstitutionTemplateLiteral` instead.

I was using astExplorer and find it odd that it maps to `FirstTemplateToken` as well.